R/HclustParam.R

```diff
@@ -58,7 +59,7 @@ setMethod(".defaultScalarArguments", "HclustParam", function(x) c(callNextMethod
 
 #' @export
 #' @rdname HclustParam-class
-HclustParam <- function(metric=NULL, method=NULL, cut.fun=NULL, cut.dynamic=FALSE, cut.height=NULL, cut.number=NULL, cut.params=list(), ...) {
+HclustParam <- function(metric=NULL, dist.fun=NULL, method=NULL, cut.fun=NULL, cut.dynamic=FALSE, cut.height=NULL, cut.number=NULL, cut.params=list(), ...) {
     if (!is.null(cut.number)) {
         .Deprecated(old="cut.number=", new="cut.params=list(k=...)")
         cut.params$k <- cut.number
```

Comment:
You could add there input check. Check that dist.fun is NULL or a function (even though other parameters seems not to have input checks...)

Reply:
I think that this is checked at the line 106 of the file AllClasses.R. 
```r
setClass("HierarchicalParam", contains=c("BlusterParam", "VIRTUAL"), 
    slots=c(
        metric="ANY",
        dist.fun="function_OR_NULL", 
        cut.fun="function_OR_NULL", 
        cut.dynamic="logical", 
        cut.params="list"
    )
)
``` 
Here it checks that the parameter is either NULL or a function. 
I can put another check if you want me to, but I feel like this is what should be done in comparison to the rest of the code.

Reply:
I am not sure what's the solution but indeed important principles are to 1) adhere to similar coding style / conventions than elsewhere in bluster package; and 2) faily often, fail early, meaning that bugs should be catched as upstream and close to their main source as possible.

Reply:
INput check is done by defining the slots, good